network/dag/notifier.go

```diff
@@ -42,6 +42,25 @@ const (
 	PayloadEventType = "payload"
 )
 
+// NewEventFatal is used by a subscriber to signal that it cannot process an Event,
+// and notification of the Event should not be retried.
+// Returns nil when err == nil.
+func NewEventFatal(err error) error {
+	if err == nil {
+		return nil
```

Comment:
I'd be very surprised to find myself with a nil value from a NewXyz function. Accidents waiting to happen. The literal looks good. We don't need a constructor. 😀 Code should be boring—not a surprise around every corner.

Reply:
That would require exporting the error on EventFatal, and implementing the error interface. I'm fine either way, but I think we need to have a larger discussion about what the pattern for our custom errors should be.